flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/scala/batch/sql/JoinITCase.scala

```diff
@@ -372,11 +366,219 @@ class JoinITCase(
 
     val sqlQuery1 = "SELECT * FROM A CROSS JOIN (SELECT count(*) FROM A HAVING count(*) < 0)"
     val result = tEnv.sql(sqlQuery1).count()
-
     Assert.assertEquals(0, result)
   }
 
   @Test
+  def testLeftNullLeftJoin (): Unit = {
```

Comment:
Several of these tests are executed with regular joins because of the restriction that for an outer single row join, the single row input must be on the inner side. Some of these tests have the single row input on the outer side which will translate them into regular joins (which is only possible because the join conditions are equality predicates). These tests can be removed, because they test the existing join implementation.

In order to enforce tests of the new outer single row joins, the join condition must be an inequality predicate. It would be good to adapt the tests queries accordingly.